.github/workflows/build-test.yml

```diff
@@ -32,23 +32,21 @@ jobs:
         working-directory: v2/
 
       - name: Build
-        run: go build .
-        working-directory: v2/cmd/nuclei/
+        run: bash build.sh retry 3 make build
+        working-directory: v2/
 
       - name: Test
         env:
           GITHUB_TOKEN: "${{ secrets.GITHUB_TOKEN }}"
-        run: go test ./...
+        run: bash build.sh retry 3 make test
         working-directory: v2/
 
```

Comment:
about unit tests failures . I think most/all of them are functionality based i.e do not make request to interactsh.
I think we can use `TestMain` for such cases . `TestMain` is package scoped so we only need to add it to package only once

```go
func TestMain(m *testing.M) {
	returnCode := m.Run()
	// retrylogic here if code != 0
}
```